CI: add Windows Integration Tests for .NET

mus65 opened the pull request:
see https://github.com/sshnet/SSH.NET/pull/1702#issuecomment-3342506642

mus65 committed:
CI: add Windows Integration Tests for .NET

see https://github.com/sshnet/SSH.NET/pull/1702#issuecomment-3342506642

mus65 committed:
fix podman setup with Windows and .NET

scott-xu commented:
How about adding some filters `--filter "Name~EcdhSha2Nist|Name~Gcm|Name~Ecdsa|Name~Tripledes"` to speed up the integration test?

mus65 commented:
I would be OK with that. We could also run these in parallel to the .NET Framework tests.

There is one test failure though, no idea about that one.

```
 Test method Renci.SshNet.IntegrationTests.ConnectivityTests.Common_CreateMoreChannelsThanMaxSessions threw exception: 
Renci.SshNet.Common.SshException: The pseudo-terminal request was not accepted by the server. Consult the server log for more information.
  Stack Trace:
     at Renci.SshNet.ShellStream..ctor(ISession session, String terminalName, UInt32 columns, UInt32 rows, UInt32 width, UInt32 height, IDictionary`2 terminalModeValues, Int32 bufferSize) in /_/src/Renci.SshNet/ShellStream.cs:line 93
   at Renci.SshNet.ServiceFactory.CreateShellStream(ISession session, String terminalName, UInt32 columns, UInt32 rows, UInt32 width, UInt32 height, IDictionary`2 terminalModeValues, Int32 bufferSize) in /_/src/Renci.SshNet/ServiceFactory.cs:line 180
   at Renci.SshNet.SshClient.CreateShellStream(String terminalName, UInt32 columns, UInt32 rows, UInt32 width, UInt32 height, Int32 bufferSize, IDictionary`2 terminalModeValues) in /_/src/Renci.SshNet/SshClient.cs:line 287
   at Renci.SshNet.SshClient.CreateShellStream(String terminalName, UInt32 columns, UInt32 rows, UInt32 width, UInt32 height, Int32 bufferSize) in /_/src/Renci.SshNet/SshClient.cs:line 279
   at Renci.SshNet.IntegrationTests.ConnectivityTests.Common_CreateMoreChannelsThanMaxSessions() in D:\a\SSH.NET\SSH.NET\test\Renci.SshNet.IntegrationTests\ConnectivityTests.cs:line 47
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

Rob-Hague committed:
Merge branch 'develop' into windows-net-tests

Rob-Hague committed:
debug

Rob-Hague committed:
x

Rob-Hague committed:
x

Rob-Hague committed:
x

Rob-Hague committed:
Merge branch 'develop' into windows-net-tests

Rob-Hague committed:
revert

mus65 commented:
I can reproduce the test failure locally on linux when running the net9.0 tests multiple times against the same SSH server container. Recreating the SSH container fixes it. So this appears to be caused by some left over state in the SSH server from the previous run.

Shouldn't we run these in in separate jobs in CI anyway? Then they would use their own SSH containers and run in parallel which would speed things up significantly.

There is obviously some kind of issue in the tests though either way. I will try to narrow it down further...

Rob-Hague commented:
Yeah, they should be separate legs. Thanks for digging in

mus65 committed:
Run Windows .NET tests in separate job

so they run in parallel and we avoid the Common_CreateMoreChannelsThanMaxSessions test failure.

mus65 committed:
Merge remote-tracking branch 'upstream/develop' into windows-net-tests

mus65 committed:
fix missing PermitTTY in RemoteSshdConfig Reset

this fixes a test failure in Common_CreateMoreChannelsThanMaxSessions
when running the tests multiple times against the same SSH server
instance.

see https://github.com/sshnet/SSH.NET/pull/1704#issuecomment-3343210311

mus65 committed:
fix coverlet artifacts

mus65 commented:
Opened #1710 to fix the test failure.

mus65 committed:
speed up Windows tests

turns out this is caused by DNS resolution taking about
2 seconds on every new connection...

mus65 commented:
Merged #1710 here because this maybe also causes the Ssh_CreateShellNoTerminal test failure.

I also figured out why the Windows tests are so slow, it's DNS....  e2c0ed247f960a48c56aab72424fba7e6b557a54 . Should be done in <15min now.

Rob-Hague commented on .github/workflows/build.yml:
add the windows integration legs here as well?

Rob-Hague reviewed (commented):
Nice!

mus65 committed:
add windows integration tests to Publish needs:
